docs/guides/migration.md

```diff
@@ -0,0 +1,223 @@
+
+    ***Old Configuration***
+    ```hcl
+    resource "aci_bridge_domain" "terraform_bd" {
+        tenant_dn = aci_tenant.terraform_tenant.id
+        name = "terraform_bd"
+        ll_addr = "::"
+        unk_mcast_act = "flood"
+    }
+    ```
+
+    ***New Configuration***
+    ```hcl
+    resource "aci_bridge_domain" "terraform_bd" {
+        parent_dn = aci_tenant.terraform_tenant.id
+        name = "terraform_bd"
+        link_local_ipv6_address = "::"
+        l3_unknown_multicast_flooding = "flood"
+    }
+    ```
+
+    The `terraform plan` command should not display any warnings anymore.
+
+    ```bash
+    No changes. Your infrastructure matches the configuration.
+
+    Terraform has compared your real infrastructure against your configuration and found no differences, so no changes are needed.
+    ```
+
+### Step 6: Cleanup
+
+1. **Remove Backup (Optional)**: If everything is working correctly, you can remove the backup state file.
+
+   ```bash
+   rm terraform.tfstate.backup
+   ```
+
+By following these steps, you can safely upgrade the ACI Terraform provider to a new version while ensuring that you have a backup of your current state in case anything goes wrong.
+
+## Changes to the ACI Provider
+
+In this section, we outline the key changes made to the Terraform ACI provider as part of the migration to the Terraform Plugin Framework. These changes aim to enhance the provider's functionality, performance, and usability. The new provider is generated from meta files, ensuring consistency and accuracy across resources and datasources. Below, we detail the specific improvements and modifications implemented during this migration.
+
+## Cleaning and Renaming of Attributes
+
+Over time, some attributes became outdated, were inconsistently named, or were not exposed correctly. This made the resources sometimes challenging to understand and thus use the ACI provider effectively. By cleaning and renaming these attributes, we aim to:
+
+1. **Improve Clarity**: Ensure that attribute names are clear and descriptive, making it easier for users to understand their purpose.
+2. **Enhance Consistency**: Standardize attribute names and structures across resources and datasources.
+3. **Simplify Usage**: Remove outdated attributes to streamline the configuration process.
+
+We understand the importance of maintaining backward compatibility to avoid disrupting existing configurations. This is why during a transition period, both old and new style attributes can be used. It allows users to continue using their existing configurations while gradually adopting the new style attributes. We have implemented deprecation warnings for old style attributes, informing users of the upcoming changes and encouraging them to transition to the new attribute style as soon as possible.
+
+> For the same ACI property, the old and new style attributes cannot be used together. Doing so will cause the provider to error during configuration validation.
```

Comment:
Would be good to change this "new style attributes"

Reply:
Yes think we as the team should decide on how to name this, have no preference will change accordingly everywhere when decision is made. 

@sajagana @samiib @anvitha-jain @gmicol @lhercot , any suggestions?

Reply:
Example:
![Screenshot 2025-03-25 at 10 27 17 AM](https://github.com/user-attachments/assets/247a0fb9-6222-4f28-ae12-11571d0c93bf)

Reply:
Can we use "legacy and modern/new attributes" to replace the "old and new style attributes"?

Reply:
putting suggestion of @shrsr here since this was in separate comment new style attributes -> appropriately redefined attributes?

Reply:
changed